components/collector/src/source_collectors/owasp_zap/security_warnings.py

```diff
@@ -14,32 +14,40 @@ class OWASPZAPSecurityWarnings(XMLFileSourceCollector):
     """Collector to get security warnings from OWASP ZAP."""
 
     async def _parse_entities(self, responses: SourceResponses) -> Entities:
-        """Override to parse the securty warnings from the XML."""
+        """Override to parse the security warnings from the XML."""
         entities: dict[str, Entity] = {}
         tag_re = re.compile(r"<[^>]*>")
         risks = cast(list[str], self._parameter("risks"))
+        count_alert_instances = cast(str, self._parameter("alerts")) == "alert instances"
         for alert in await self.__alerts(responses, risks):
             ids = [
                 alert.findtext(id_tag, default="") for id_tag in ("alert", "pluginid", "cweid", "wascid", "sourceid")
             ]
             name = alert.findtext("name", default="")
             description = tag_re.sub("", alert.findtext("desc", default=""))
             risk = alert.findtext("riskdesc", default="")
-            for alert_instance in alert.findall("./instances/instance"):
-                method = alert_instance.findtext("method", default="")
-                uri = self.__stable(hashless(URL(alert_instance.findtext("uri", default=""))))
-                key = md5_hash(f"{':'.join(ids)}:{method}:{uri}")
-                entities[key] = Entity(
-                    key=key,
-                    old_key=md5_hash(f"{':'.join(ids[1:])}:{method}:{uri}"),
-                    name=name,
-                    description=description,
-                    uri=uri,
-                    location=f"{method} {uri}",
-                    risk=risk,
-                )
+            entity_kwargs = dict(name=name, description=description, risk=risk)
+            if count_alert_instances:
+                for alert_instance in alert.findall("./instances/instance"):
+                    entity = self.__alert_entity(ids, entity_kwargs, alert_instance)
+                    entities[entity["key"]] = entity
+            else:
+                entity = self.__alert_entity(ids, entity_kwargs)
+                entities[entity["key"]] = entity
         return list(entities.values())
 
+    def __alert_entity(self, ids, entity_kwargs, alert_instance=None) -> Entity:
```

Comment:
maybe make this two functions? one alert_entity_type and one alert_entity, In the end these are very different entities.

Reply:
You're right. I will change this.